test/api.js

```diff
@@ -296,46 +308,100 @@ describe('/avatar', function() {
+          scope: ['profile:avatar:write']
+        });
+        return Server.api.delete({
+          url: '/avatar/' + id,
+          headers: {
+            authorization: 'Bearer ' + tok,
+          }
+        }).then(function(res) {
+          assert.equal(res.statusCode, 401);
+        });
       });
-      return Server.api.delete({
-        url: '/avatar/' + id,
-        headers: {
-          authorization: 'Bearer ' + tok,
-        }
-      }).then(function(res) {
-        assert.equal(res.statusCode, 401);
+
+      it('should remove avatar from user', function() {
+        mock.token({
+          user: user,
+          email: 'user@example.domain',
+          scope: ['profile:avatar:write']
+        });
+        return Server.api.delete({
+          url: '/avatar/' + id,
+          headers: {
+            authorization: 'Bearer ' + tok,
+          }
+        }).then(function(res) {
+          assert.equal(res.statusCode, 200);
+          return db.getAvatar(id);
+        }).then(function(avatar) {
+          assert.equal(avatar, undefined);
+        });
       });
     });
 
-    it('should remove avatar from user', function() {
-      mock.token({
-        user: user,
-        email: 'user@example.domain',
-        scope: ['profile:avatar:write']
+    describe('uploaded', function() {
+      var s3url;
+      var id;
+      before(function() {
+        this.slow(2000);
+        this.timeout(3000);
+        mock.token({
+          user: user,
+          email: 'user@example.domain',
+          scope: ['profile:avatar:write']
+        });
+        mock.image();
+        return Server.api.post({
+          url: '/avatar/upload',
+          payload: imageData,
+          headers: { authorization: 'Bearer ' + tok,
+            'content-type': 'image/png',
+            'content-length': imageData.length
+          }
+        }).then(function(res) {
+          assert.equal(res.statusCode, 201);
+          s3url = res.result.url;
+          id = res.result.id;
+        });
       });
-      return Server.api.delete({
-        url: '/avatar/' + id,
-        headers: {
-          authorization: 'Bearer ' + tok,
-        }
-      }).then(function(res) {
-        assert.equal(res.statusCode, 200);
-        return db.getAvatar(id);
-      }).then(function(avatar) {
-        assert.equal(avatar, undefined);
+
+      it('should remove avatar from user and s3', function() {
+        mock.token({
+          user: user,
+          email: 'user@example.domain',
+          scope: ['profile:avatar:write']
+        });
+        mock.deleteImage();
+        return Server.api.delete({
+          url: '/avatar/' + id,
+          headers: {
+            authorization: 'Bearer ' + tok,
+          }
+        }).then(function(res) {
+          assert.equal(res.statusCode, 200);
+          return db.getAvatar(id);
+        }).then(function(avatar) {
+          assert.equal(avatar, undefined);
+          return Static.get(s3url);
```

Comment:
IIUC, this is the part where we check whether the image has been deleted from the mocked S3.  We should add a check in the preceeding test to ensure that the mock worked correctly, i.e. check that we can `Static.get(s3url)` and receive a `200 OK`, to distinguish between "it was deleted successully" and "it was never uploaded in the first place".